analyzers/tests/SonarAnalyzer.UnitTest/CFG/Sonar/SonarControlFlowGraphTest.cs

```diff
@@ -365,12 +364,12 @@ public void Cfg_If_Else()
             var falseBlock = cfg.Blocks.ToList()[2];
             var exitBlock = cfg.ExitBlock;
 
-            branchBlock.SuccessorBlocks.Should().OnlyContainInOrder(trueBlock, falseBlock);
+            branchBlock.SuccessorBlocks.Should().BeEquivalentTo(new[] { trueBlock, falseBlock });
```

Comment:
here you use instead of `OnlyContainInOrder` the `BeEquivalentTo` , but above you used 

```
body.OperationsAndBranchValue.Should().Equal(body.Operations[0], body.BranchValue);
```

what's the difference?

Reply:
Equal => Same elements in the same order
BeEquivalentTo => Same elements. Order doesn't matter.

The problem with operations is that the order is not guaranteed, so BeEquivalentTo is almost always the right thing to do. You can have tests succeed locally and fail on CI if you check the order as well. When in doubt, I used `BeEquivalentTo` over `Equals`. `Equals` was used for single element checks because there is no order and `Equals` is more concise.

Reply:
(discussed offline - need clarification from  Pavel about the importance of the order in the UTs)

Reply:
@pavel-mikula-sonarsource I relaxed the ordering constraint on some of the cfg tests (block ordering). Is it okay with you or should I keep the order requirement if it is already there? My reasoning is described in the comment above https://github.com/SonarSource/sonar-dotnet/pull/6627#discussion_r1092101738

Reply:
For `SuccessorBlocks`, we don't care about the order. For these tests, it is important to make sure that we shim the correct properties and that they contain the expected data (whatever Roslyn provides).

A question to consider could be if we really needed to do this change in this PR. As the change is already done and there's no harm, I don't want to discuss it.

I'll let @andrei-epure-sonarsource resolve this thread as I'm not a reviewer here.